R/constructor.R

```diff
@@ -48,7 +48,7 @@ constructor_args <- function(parent, properties = list()) {
 
   self_args <- names2(properties)
   # Remove dynamic arguments
-  self_args <- self_args[vlapply(properties, function(x) is.null(x$getter))]
+  self_args <- self_args[vlapply(properties, function(x) is.null(x$getter) || !is.null(x$default))]
```

Comment:
Note that a property with only a getter could be computed entirely dynamically, so we wouldn't want it in the constructor. I still think we should use the prototype approach, where the object just starts with its defaults, and there's no need to set a read-only property during construction. 

Btw, there is actually a bug here, where a property with a setter (and a getter) is dropped, in error. This is addressed by the prop_is_read_only branch.

Reply:
What should the interface be for someone looking to implement a class with a property that can be set when the instance is initially constructed but is read-only after that?

Reply:
By the way, this looks good to me: https://github.com/RConsortium/S7/compare/main...prop_is_read_only

Reply:
See issue #404 for my proposed approach to construct-only properties.